src/shop.c

```diff
@@ -0,0 +1,1368 @@
+        BuyMenuDrawGraphics();
+        BuyMenuAddScrollIndicatorArrows();
+        taskId = CreateTask(Task_BuyMenu, 8);
+        gTasks[taskId].tListTaskId = ListMenuInit(&gMultiuseListMenuTemplate, 0, 0);
+        BlendPalettes(0xFFFFFFFF, 0x10, RGB_BLACK);
+        BeginNormalPaletteFade(0xFFFFFFFF, 0, 0x10, 0, RGB_BLACK);
+        SetVBlankCallback(VBlankCB_BuyMenu);
+        SetMainCallback2(CB2_BuyMenu);
+        break;
+    }
+}
+
+static bool8 InitShopData(void)
+{
+    gShopTilemapBuffer1 = Alloc(sizeof(*gShopTilemapBuffer1));
+    if (gShopTilemapBuffer1 == 0)
+        goto CANCEL;
+    
+    gShopTilemapBuffer2 = Alloc(sizeof(*gShopTilemapBuffer2));
+    if (gShopTilemapBuffer2 == 0)
+        goto CANCEL;
+    
+    gShopTilemapBuffer3 = Alloc(sizeof(*gShopTilemapBuffer3));
+    if (gShopTilemapBuffer3 == 0)
+        goto CANCEL;
+    
+    gShopTilemapBuffer4 = Alloc(sizeof(*gShopTilemapBuffer4));
+    if (gShopTilemapBuffer4 == 0)
+        goto CANCEL;
+    return TRUE;
+    
+    CANCEL:
+    BuyMenuFreeMemory();
+    SetShopExitCallback();
+    return FALSE;
+}
+
+static void BuyMenuInitBgs(void)
+{
+    ResetBgsAndClearDma3BusyFlags(0);
+    InitBgsFromTemplates(0, sShopBuyMenuBgTemplates, NELEMS(sShopBuyMenuBgTemplates));
+    SetBgTilemapBuffer(1, gShopTilemapBuffer2);
+    SetBgTilemapBuffer(2, gShopTilemapBuffer4);
+    SetBgTilemapBuffer(3, gShopTilemapBuffer3);
+    SetGpuReg(0x10, 0);
+    SetGpuReg(0x12, 0);
+    SetGpuReg(0x14, 0);
+    SetGpuReg(0x16, 0);
+    SetGpuReg(0x18, 0);
+    SetGpuReg(0x1A, 0);
+    SetGpuReg(0x1C, 0);
+    SetGpuReg(0x1E, 0);
+    SetGpuReg(0x50, 0);
+    SetGpuReg(0, 0x1040);
+    ShowBg(0);
+    ShowBg(1);
+    ShowBg(2);
+    ShowBg(3);
+}
+
+static void BuyMenuDecompressBgGraphics(void)
+{
+    void* pal;
+    
+    DecompressAndCopyTileDataToVram(1, gBuyMenuFrame_Gfx, 0x480, 0x3DC, 0);
+    if ((gShopData.martType) != MART_TYPE_TMHM)
+        LZDecompressWram(gBuyMenuFrame_Tilemap, gShopTilemapBuffer1);
+    else
+        LZDecompressWram(gBuyMenuFrame_TmHmTilemap, gShopTilemapBuffer1);
+    
+    pal = Alloc(0x40);
+    LZDecompressWram(gBuyMenuFrame_Pal, pal);
+    LoadPalette(pal, 0xB0, 0x20);
+    LoadPalette(pal + 0x20, 0x60, 0x20);
+    Free(pal);
+}
+
+static void sub_809B10C(bool32 a0)
+{
+    u8 v;
+    
+    if (a0 == FALSE)
+        v = 0xB;
+    else 
+        v = 6;
+    
+    if ((gShopData.martType) != MART_TYPE_TMHM)
+        SetBgRectPal(1, 0, 0xE, 0x1E, 6, v);
+    else
+        SetBgRectPal(1, 0, 0xC, 0x1E, 8, v);
+    
+    ScheduleBgCopyTilemapToVram(1);
+}
+
+static void BuyMenuDrawGraphics(void)
+{
+    sub_809B764();
+    BuyMenuCopyTilemapData();
+    BuyMenuDrawMoneyBox();
+    ScheduleBgCopyTilemapToVram(0);
+    ScheduleBgCopyTilemapToVram(1);
+    ScheduleBgCopyTilemapToVram(2);
+    ScheduleBgCopyTilemapToVram(3);
+}
+
+static bool8 BuyMenuBuildListMenuTemplate(void)
+{
+    u16 i, v;
+    struct ListMenuItem **list = &sShopMenuListMenu;
```

Comment:
This looks weird. Are you sure it's necessary?

Reply:
probably not, but it worked at the time lol.  Here's the compiler explorer: https://cexplore.karathan.at/z/aKHXv1

Reply:
https://cexplore.karathan.at/z/KuVZTF